src/components/Settings/TransactionDeadlineSettings/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { Trans } from '@lingui/macro'
+import Expand from 'components/Expand'
+import QuestionHelper from 'components/QuestionHelper'
+import Row from 'components/Row'
+import { Input, InputContainer } from 'components/Settings/Input'
+import { DEFAULT_DEADLINE_FROM_NOW } from 'constants/misc'
+import ms from 'ms.macro'
+import React, { useState } from 'react'
+import { useUserTransactionTTL } from 'state/user/hooks'
+import { ThemedText } from 'theme'
+
+enum DeadlineError {
+  InvalidInput = 'InvalidInput',
+}
+
+const THREE_DAYS_IN_SECONDS = ms`3 days` / 1000
+const NUMBERS_ONLY = /^[0-9\b]+$/
+
+export default function TransactionDeadlineSettings() {
+  const [deadline, setDeadline] = useUserTransactionTTL()
+
+  const defaultInputValue = deadline && deadline !== DEFAULT_DEADLINE_FROM_NOW ? (deadline / 60).toString() : ''
```

Comment:
why dont we just store deadline as minute-based and then just convert to `second` when we submit the tx 🤔  maybe outside the scope of this PR tho lol

Reply:
Responded in another comment, marking as a TODO.